internal/tools/pipeline/providers/reconciler/task_reconciler.go

```diff
@@ -105,6 +105,11 @@ func (tr *defaultTaskReconciler) ReconcileOneTaskUntilDone(ctx context.Context,
 			}
 		}
 
+		// overwrite current task with latest, otherwise aop will lose result information and so on
+		if err := tr.overwriteTaskWithLatest(task); err != nil {
```

Comment:
put this invoke inside tr.TeardownAfterReconcileDone (internal/tools/pipeline/providers/reconciler/task_reconciler.go:320)